src/handler.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+var uniq = require('lodash.uniq');
+
+function Handler(retriever) {
+  this._fetched = false;
+  this._page = 1;
+  this._initRequest = false;
+  this._retriever = retriever;
+  this._totalPages = 1;
+  this.hits = [];
+  this.lastPage = false;
+}
+
+Handler.prototype._handleResponse = function(response) {
+  this.response = response;
+  var data = response.body;
+
+  this.totalHits = data.totalHits;
+  this.hits = uniq(this.hits.concat(data.hits));
+
+  var per_page = this._retriever._query.per_page || 20;
+  this._totalPages = Math.ceil(this.totalHits / per_page);
+
+  if (this._page === this._totalPages) {
+    this.lastPage = true;
+  }
+
+  return response;
+};
+
+Handler.prototype.next = function() {
```

Comment:
Would it make sense to remove `get`? It might make more sense to treat it as a generator than not. The methods would be used to set up the query params and then calling `next` would return a promise for the initial request. Calling it again would return a promise for the next page, etc.